src/transaction.ts

```diff
@@ -602,6 +616,15 @@ export class Snapshot extends EventEmitter {
       }
     );
 
+    const headers = this.resourceHeader_;
+    if (
+      (this.session.parent.parent.parent as Spanner).routeToLeaderEnabled &&
+      (this._options.readWrite !== undefined ||
```

Comment:
Ah... I guess that this is the reason that the if statement is not part of the `addLeaderAwareRoutingHeader` function.

Reply:
Yes, I have not moved if condition to addLeaderAwareRoutingHeader